### ISSUE TITLE
📝 Incorrect trailing comma added with `trailingComma: all` and `arrowParentheses: asNeeded`

### ISSUE THREAD

**Issue by Aloso** [biomejs/biome#831]
Biome may sometimes add a trailing comma that breaks the code (producing invalid syntax)!

### Environment information

```bash
CLI:
  Version:                      1.3.3
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v18.18.2"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### Configuration

```JSON
{
  "$schema": "https://biomejs.dev/schemas/1.3.3/schema.json",
  "formatter": {
    "enabled": true,
    "indentWidth": 2,
    "indentStyle": "space",
    "lineWidth": 100
  },
  "javascript": {
    "formatter": {
      "trailingComma": "all",
      "arrowParentheses": "asNeeded",
      "quoteStyle": "single",
      "semicolons": "always"
    }
  }
}
```

### Playground link

https://biomejs.dev/playground/?lineWidth=100&indentStyle=space&quoteStyle=single&arrowParentheses=as-needed&code=aQBtAHAAbwByAHQAIAB0AHkAcABlACAAewAgAFQAdwBpAGwAaQBvAFYAaQBkAGUAbwBUAHIAYQBjAGsAIAB9ACAAZgByAG8AbQAgACcALgAuAC8AaQBuAHQAZQByAGYAYQBjAGUAcwAvAFQAdwBpAGwAaQBvAFYAaQBkAGUAbwBUAHIAYQBjAGsAJwA7AAoAaQBtAHAAbwByAHQAIAB0AHkAcABlACAAewAgAFYAaQBkAGUAbwBUAHIAYQBjAGsAIAB9ACAAZgByAG8AbQAgACcALgAuAC8AaQBuAHQAZQByAGYAYQBjAGUAcwAvAFYAaQBkAGUAbwBUAHIAYQBjAGsAJwA7AAoACgBlAHgAcABvAHIAdAAgAGMAbwBuAHMAdAAgAGMAcgBlAGEAdABlAFYAaQBkAGUAbwBUAHIAYQBjAGsAOgAgACgACgAgACAAdAB3AGkAbABpAG8AVgBpAGQAZQBvAFQAcgBhAGMAawA6ACAAVAB3AGkAbABpAG8AVgBpAGQAZQBvAFQAcgBhAGMAawAsAAoAKQAgAD0APgAgAFYAaQBkAGUAbwBUAHIAYQBjAGsAIAA9ACAAdAB3AGkAbABpAG8AVgBpAGQAZQBvAFQAcgBhAGMAawAgAD0APgAgACgAewAKACAAIABpAG4AcwB0AGEAbgBjAGUAOgAgAHQAdwBpAGwAaQBvAFYAaQBkAGUAbwBUAHIAYQBjAGsALAAKAH0AKQA7AAoA

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
This seems to be fixed on main.
